### PR TITLE
Add SafeCmd.run_sync with full parity tests

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Implement SafeCmd.run_sync to drive the event loop synchronously with identical semantics to run()
- Add comprehensive unit tests for parity and edge-cases
- Update docs to reflect the synchronous API and usage

### Core Functionality
- Implemented SafeCmd.run_sync in cuprum/sh.py using asyncio.run(self.run(...)) to mirror async behavior and identical return semantics (capture, echo, context).

### Tests
- Added extensive unit tests in cuprum/unittests/test_safe_cmd_run.py to verify:
  - run_sync captures stdout/stderr and exit code by default
  - Capturing and stderr-only scenarios
  - echo=True echoes output while capturing
  - env overrides via ExecutionContext are correctly applied without leaking to global env
  - Non-zero exit codes are captured and ok flag is set accordingly
  - cwd override works as expected
  - capture=False disables capturing stdout/stderr
  - Echoing to custom sinks via ExecutionContext stdout_sink and stderr_sink
  - Decoding with configured encoding and error handling
- Extended behaviour tests:
  - tests/behaviour/test_execution_runtime.py: added scenario to cover synchronous run captures output by default
  - tests/features/execution_runtime.feature: added corresponding feature scenario

### Documentation
- docs/roadmap.md: mark run_sync parity as implemented (x)
- docs/users-guide.md: document synchronous execution with run_sync and its parity with run()

### Test Plan
- Run unit tests for SafeCmd run_sync parity (e.g., pytest cuprum/unittests/test_safe_cmd_run.py -q)
- Run full test suite (pytest) including tests related to execution runtime and behaviour
- Optionally run behave to execute updated behaviours/features (tests/features/execution_runtime.feature)

🌿 Generated by [Terry](https://www.terragonlabs.com)

---

ℹ️ Tag @terragon-labs to ask questions and address PR feedback

📎 **Task**: https://www.terragonlabs.com/task/9122a1b4-abf8-4c90-a8a1-c87d1e77cf7e

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[cuprum/unittests/test_safe_cmd_run.py](https://github.com/leynos/cuprum/pull/10/files/848382f202ac6e3ca6803726675d283ff8185489#diff-369c2ef11857236ab221001162c4883c68bf5308a10e6c8cb0248e2b6306ac79)

Comment on lines +442 to +461

~~~python
def test_run_sync_decodes_with_configured_encoding(
    python_builder: typ.Callable[..., SafeCmd],
) -> None:
    """run_sync() uses the configured encoding/errors when decoding output."""
    command = python_builder(
        "-c",
        ("import sys; sys.stdout.buffer.write(bytes([0x96])); sys.stdout.flush()"),
    )

    result = command.run_sync(
        context=ExecutionContext(
            encoding="cp1252",
            errors="strict",
        ),
    )

    assert result.exit_code == 0
    assert result.ok is True
    assert result.stdout == "\u2013"
    assert result.stderr == ""
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/74471/delta?repo-id=1354509&review-id=10&biomarker=Code+Duplication&filename=cuprum%2Funittests%2Ftest_safe_cmd_run.py&method=)
The module contains 12 functions with similar structure: test_run_applies_cwd_override,test_run_applies_env_overrides,test_run_captures_and_echoes_stderr,test_run_captures_stderr_only and 8 more functions

File: cuprum/unittests/test_safe_cmd_run.py [hunk 1]
```diff
@@ -13,41 +13,66 @@
...
 from cuprum import ECHO, sh
-from cuprum.sh import ExecutionContext
+from cuprum.sh import CommandResult, ExecutionContext
 from tests.helpers.catalogue import python_builder as build_python_builder
 
 if typ.TYPE_CHECKING:
     from cuprum.sh import SafeCmd
 
+type ExecuteFn = typ.Callable[[SafeCmd, dict[str, typ.Any]], CommandResult]
+
```
[the remark this answers is not in the document]
Comment (leynos) [reply cuprum/unittests/test_safe_cmd_run.py:23]: @coderabbitai I think Codex is talking mince here. What say you?

I guess we could put `typ.Callable[[SafeCmd, dict[str, typ.Any]], CommandResult]` in quotes, but CI is succeeding.
